jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/Handler.java

```diff
@@ -692,7 +692,15 @@ public Wrapper(Handler handler)
         public Wrapper(boolean dynamic, Handler handler)
         {
             super(dynamic);
-            _handler = handler == null ? null : Singleton.updateHandler(this, handler);
+            _handler = handler;
+            addBean(handler);
```

Comment:
So this is where we have our constructor escape. Can't this `addBean` call be moved to `doStart`?

Reply:
The `addBean(handler)` is part of the call to `Singleton.updateHandler(this, _handler);` in `doStart()`.

But that is too late to wire up the LifeCycle for start reasons.
The resulting handler never gets started if we wait till `doStart()` to do it.
We are then having to manage lifecycle start/stop manually, which is worse.

Reply:
With the old code, where we have updateHandler in the constructor, such as on the released 12.0.5, we have this same vhosts is null issue anyway.

See issue #11220

```
SLF4J(E): Failed toString() invocation on an object of type [org.eclipse.jetty.server.handler.ContextHandler]
SLF4J(E): Reported exception:
java.lang.NullPointerException: Cannot invoke "java.util.List.stream()" because "this._vhosts" is null
	at org.eclipse.jetty.server.handler.ContextHandler.getVirtualHosts(ContextHandler.java:391)
	at org.eclipse.jetty.server.handler.ContextHandler.toString(ContextHandler.java:1032)
	at org.slf4j.helpers.MessageFormatter.safeObjectAppend(MessageFormatter.java:291)
	at org.slf4j.helpers.MessageFormatter.deeplyAppendParameter(MessageFormatter.java:263)
	at org.slf4j.helpers.MessageFormatter.arrayFormat(MessageFormatter.java:225)
	at org.slf4j.helpers.MessageFormatter.arrayFormat(MessageFormatter.java:160)
	at ch.qos.logback.classic.spi.LoggingEvent.getFormattedMessage(LoggingEvent.java:397)
	at ch.qos.logback.classic.spi.LoggingEvent.prepareForDeferredProcessing(LoggingEvent.java:253)
	at ch.qos.logback.core.OutputStreamAppender.subAppend(OutputStreamAppender.java:226)
	at ch.qos.logback.core.OutputStreamAppender.append(OutputStreamAppender.java:102)
	at ch.qos.logback.core.UnsynchronizedAppenderBase.doAppend(UnsynchronizedAppenderBase.java:85)
	at ch.qos.logback.core.spi.AppenderAttachableImpl.appendLoopOnAppenders(AppenderAttachableImpl.java:51)
	at ch.qos.logback.classic.Logger.appendLoopOnAppenders(Logger.java:272)
	at ch.qos.logback.classic.Logger.callAppenders(Logger.java:259)
	at ch.qos.logback.classic.Logger.buildLoggingEventAndAppend(Logger.java:426)
	at ch.qos.logback.classic.Logger.filterAndLog_2(Logger.java:419)
	at ch.qos.logback.classic.Logger.debug(Logger.java:495)
	at org.eclipse.jetty.util.component.ContainerLifeCycle.addBean(ContainerLifeCycle.java:419)
	at org.eclipse.jetty.util.component.ContainerLifeCycle.addBean(ContainerLifeCycle.java:317)
	at org.eclipse.jetty.util.component.ContainerLifeCycle.updateBean(ContainerLifeCycle.java:802)
	at org.eclipse.jetty.server.Handler$Singleton.updateHandler(Handler.java:397)
	at org.eclipse.jetty.server.Handler$Wrapper.<init>(Handler.java:695)
	at org.eclipse.jetty.server.Handler$Wrapper.<init>(Handler.java:682)
	at org.eclipse.jetty.server.handler.ContextHandler.<init>(ContextHandler.java:177)
	at org.eclipse.jetty.server.handler.ContextHandler.<init>(ContextHandler.java:167)
	at worktrack.WorktracksServer.main(WorktracksServer.java:41)
```